docs/tutorials/runtime-module-in-own-crate.md

```diff
@@ -0,0 +1,167 @@
+[dependencies.parity-codec]
+default-features = false
+features = ['derive']
+version = '3.5'
+
+# These dependencies specify a git branch
+[dependencies.support]
+default_features = false
+git = 'https://github.com/paritytech/substrate.git'
+package = 'srml-support'
+branch = 'v1.0'
+
+[dependencies.system]
+default_features = false
+git = 'https://github.com/paritytech/substrate.git'
+package = 'srml-system'
+branch = 'v1.0'
+# Could also use a specific revision
+# rev = '<commit hash>'
+```
+
+The final section of the `Cargo.toml` file specifies the dev dependencies, and is not necessary to make the template compile. Remove that final section to confirm this for yourself.
+
+At this point we have a runtime module that compiles. It comes with a test, so let's see if that test passes.
+```bash
+cargo test
+```
+
+Turns out we'll need some additional dependencies to make the test pass. You can see this from the error messages, or by looking at the test module itself. Let's add `primitives`, `runtime-primitives`, and `runtime-io` as dev dependencies. The dev dependencies do not need to be listed as part of the `std` feature because they will not be part of the release.
+
+Add these lines back to your `Cargo.toml`.
+```toml
+
+[dev-dependencies.primitives]
+default_features = false
+git = 'https://github.com/paritytech/substrate.git'
+package = 'substrate-primitives'
+branch = 'v1.0'
+
+[dev-dependencies.runtime-primitives]
+default_features = false
+git = 'https://github.com/paritytech/substrate.git'
+package = 'sr-primitives'
+branch = 'v1.0'
+
+[dev-dependencies.runtime-io]
+default_features = false
+git = 'https://github.com/paritytech/substrate.git'
+package = 'sr-io'
+branch = 'v1.0'
+```
+
+It will take a little experimentation to gain familiarity with what dependencies you need for each module you dream up. So the take-away from this section is not that `Cargo.toml` should always look like it does in the template. Rather, `Cargo.toml` needs to have the correct dependencies and you now know how to specify them.
+
+## Add your Runtime Module to your Node
+With our runtime module now compiling and passing tests (well, passing test anyway) we're ready to add it to our node template.
+
+First we need to add our newly-created crate as a dependency in the node runtime's `Cargo.toml`.
+
+```toml
+[dependencies.test_module]
+default_features = false
+path = "../../test_module"
+# Optionally specify a version
+# version = "0.1.0"
+```
+
+And just as before, we need to tell the module to only build its `std` feature when the runtime itself does.
+```toml
+std = [
+  ...
+  'test_module/std',
+]
+```
+
+Next we'll update `runtime/lib.rs` in modular-chain to actually use the new runtime module. The lines that bring in the template module that is coded directly in the node runtime are still there, so we just need to update them in three places.
+
+1. Remove `mod template` as we've already brought that dependency in via `Cargo.toml`
```

Comment:
The user does not need to remove the template module for this to work.

Reply:
I agree that it isn't strictly necessary. But I think it's wise for two reasons.
1. In general it's better to remove excess cruft than leave it around.
2. I'm trying to illustrate how things look _different_ between writing your module as part of your runtime vs writing it as an external crate. Removing that line emphasizes that we do that step differently now. It emphasizes that we do the stuff in cargo.toml _instead_ of `mod template`.

Reply:
I still kind of disagree here. You have not for example removed the `template.rs` file from your project, and you say to "update" the definitions rather than simply write new ones.

I think the more common pattern for a user adding/creating a new module is that they would leave everything about their existing node the same, and simply add the new node on top of it.

If you want to make a statement about the `mod` thing you can, but also this is a little bit implied with the use of all the other modules used in the Substrate Node Template (and any minimal familiarity with Rust)